mailbox/store/src/test/java/org/apache/james/mailbox/store/mail/model/ListMailboxAssert.java

```diff
@@ -0,0 +1,97 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.store.mail.model;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
+import org.apache.james.mailbox.model.MailboxId;
+import org.apache.james.mailbox.store.mail.model.impl.SimpleMailbox;
+
+import com.google.common.base.Function;
+import com.google.common.collect.FluentIterable;
+
+public class ListMailboxAssert {
+
+    private final List<Mailbox> actual;
+
+
+    private static final List<MailboxId> toIds(List<SimpleMailbox> expecteds) {
+        return FluentIterable.from(expecteds).transform(new Function<Mailbox, MailboxId>() {
+            @Override
+            public MailboxId apply(Mailbox input) {
+                return input.getMailboxId();
+            }
+            
+        }).toList();
+    }
+
+    private static final List<String> toUsers(List<SimpleMailbox> expecteds) {
+        return FluentIterable.from(expecteds).transform(new Function<Mailbox, String>() {
+            @Override
+            public String apply(Mailbox input) {
+                return input.getUser();
+            }
+            
+        }).toList();
+    }
+
+    private static final List<String> toNames(List<SimpleMailbox> expecteds) {
+        return FluentIterable.from(expecteds).transform(new Function<Mailbox, String>() {
+            @Override
+            public String apply(Mailbox input) {
+                return input.getName();
+            }
+            
+        }).toList();
+    }
+
+    private static final List<String> toNamespaces(List<SimpleMailbox> expecteds) {
+        return FluentIterable.from(expecteds).transform(new Function<Mailbox, String>() {
+            @Override
+            public String apply(Mailbox input) {
+                return input.getNamespace();
+            }
+            
+        }).toList();
+    }
+
+    private ListMailboxAssert(List<Mailbox> actual) {
+        this.actual = actual;
+    }
+
+    public static ListMailboxAssert assertMailboxes(List<Mailbox> actual) {
+        return new ListMailboxAssert(actual);
+    }
+
+    public void contain(List<SimpleMailbox> expecteds) {
+        assertThat(actual).extracting("mailboxId").containsAll(toIds(expecteds));
+        assertThat(actual).extracting("name").containsAll(toNames(expecteds));
+        assertThat(actual).extracting("namespace").containsAll(toNamespaces(expecteds));
+    }
+
+    public void containAll(List<SimpleMailbox> expecteds) {
+        assertThat(actual).extracting("mailboxId").containsAll(toIds(expecteds));
+        assertThat(actual).extracting("user").containsAll(toUsers(expecteds));
+        assertThat(actual).extracting("name").containsAll(toNames(expecteds));
+        assertThat(actual).extracting("namespace").containsAll(toNamespaces(expecteds));
+    }
```

Comment:
With ContainsAll we can have extra entries. It's quitte weak.

Please provides containsOnly and containsExactly on you Assertions.

Moreover, Immagine : 

```
actlual = Immutablist.of(new Mailbox(id1, user1, namespace1, name1), new Mailbox(id2, user2, namespace2, name2));
expected = Immutablist.of(new Mailbox(id1, user1, namespace2, name2), new Mailbox(id2, user2, namespace1, name1));
```

Your contains methos will not fail but they should have...

What you need is to convert both mailbox lists into a common data type with your equals and hashcode. No need for it to be a Mailbox implementation. It can be an internal class. It would avoid mixing fields.